src/osd/osd_types.h

```diff
@@ -4117,7 +4117,7 @@ struct object_info_t {
   explicit object_info_t(bufferlist& bl) {
     decode(bl);
   }
-  object_info_t operator=(bufferlist& bl) {
+  object_info_t &operator=(bufferlist& bl) {
```

Comment:
and what is deleted in your change is not the default copy assignment operator. the one created by compiler by default is 
```c++
object_info_t& operator=(object_info_t&)
```